src/VisualStudio/Core/Def/Packaging/PackageSearchService.RemoteControlService.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Internal.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.Packaging
+{
+    internal partial class PackageSearchService
+    {
+        private class RemoteControlService : IPackageSearchRemoteControlService
+        {
+            private readonly object _remoteControlService;
+
+            public RemoteControlService(object remoteControlService)
+            {
+                _remoteControlService = remoteControlService;
+            }
+
+            public IPackageSearchRemoteControlClient CreateClient(string hostId, string serverPath, int pollingMinutes)
+            {
+                var serviceType = _remoteControlService.GetType();
+                var serviceAssembly = serviceType.Assembly;
+                var clientType = serviceAssembly.GetType("Microsoft.VisualStudio.Services.RemoteControl.VSRemoteControlClient");
+
+                var vsClient = Activator.CreateInstance(clientType, args: new object[]
+                {
+                    HostId,
+                    serverPath,
+                    pollingMinutes,
+                    null
+                });
+
+                var clientField = vsClient.GetType().GetField("client", BindingFlags.Instance | BindingFlags.NonPublic);
```

Comment:
Yes/no.  If we can take a dependency on their internal nuget package, then no.  We can just literally new up their public type.  However, from talking to them and Dave, it seemed tricky for us to depend on their nuget package.  So this was a suggested workaround.